content/tutorial/02-sveltekit/03-mutating-data/04-progressive-enhancement/README.md

```diff
@@ -0,0 +1,31 @@
+---
+title: Progressive Enhancement
+---
+
+So far, the `<form>` we have built works completely without JavaScript - which is great! If we have JavaScript enabled though ([which may be less often than you think](https://kryogenix.org/code/browser/everyonehasjs.html)), we can provide a better user experience by not reloading the whole page, showing loading animations, and more.
+
+The easiest way to progressively enhance our `<form>` is to add the `enhance` action to it:
+
+```svelte
+<script>
+	+++import { enhance } from '$app/forms';+++
+	export let form;
+</script>
+
+<form method="POST" action="?/login" +++use:enhance+++>
+	<input name="email" type="email">
+	<input name="password" type="password">
+	{#if form?.message}
+		<span>{form?.message}</span>
+	{/if}
+	<button>Log in</button>
+	<button formAction="?/register">Register</button>
+</form>
+```
+
+And that's all! Now, when JavaScript is enabled, `use:enhance` will emulate the browser-native behaviour, just without the full-page reloads. It will:
+
+- update the `form` property - but only if the action is on the same page you're submitting from. So for example if your form looks like <form action="/somewhere/else" ..>, form and $page will not be updated. This is because in the native form submission case you would be redirected to the page the action is on.
```

Comment:
Possibly use code formatting here:

```suggestion
- update the `form` property - but only if the action is on the same page you're submitting from. So for example if your form looks like `<form action="/somewhere/else" ..>`, `form` and `$page` will not be updated. This is because in the native form submission case you would be redirected to the page the action is on.
```